app/src/main/java/edu/wpi/first/shuffleboard/app/components/Tile.java

```diff
@@ -0,0 +1,109 @@
+package edu.wpi.first.shuffleboard.app.components;
+
+import edu.wpi.first.shuffleboard.api.components.EditableLabel;
+import edu.wpi.first.shuffleboard.api.util.PropertyUtils;
+import edu.wpi.first.shuffleboard.api.util.PseudoClassProperty;
+import edu.wpi.first.shuffleboard.api.widget.Component;
+import edu.wpi.first.shuffleboard.api.widget.Layout;
+import edu.wpi.first.shuffleboard.api.widget.TileSize;
+import edu.wpi.first.shuffleboard.api.widget.Widget;
+
+import org.fxmisc.easybind.EasyBind;
+
+import java.io.IOException;
+
+import javafx.beans.property.BooleanProperty;
+import javafx.beans.property.Property;
+import javafx.beans.property.SimpleObjectProperty;
+import javafx.fxml.FXMLLoader;
+import javafx.scene.layout.BorderPane;
+
+public class Tile<T extends Component> extends BorderPane {
+
+  private final Property<T> content = new SimpleObjectProperty<>(this, "content", null);
+
+  private final Property<TileSize> size = new SimpleObjectProperty<>(this, "size", null);
+  private final BooleanProperty selected = new PseudoClassProperty(this, "selected");
+
+  /**
+   * Creates an empty tile. The content and size must be set with {@link #setContent(T)} and
+   * {@link #setSize(TileSize)}.
+   */
+  protected Tile() {
+    try {
+      FXMLLoader loader = new FXMLLoader(Tile.class.getResource("Tile.fxml"));
+      loader.setRoot(this);
```

Comment:
Do you want to set the controller to this class also?

Reply:
no???

Reply:
No class should ever be both a component and a controller.